extensions/wikia/UserRenameTool/RenameUserProcess.class.php

```diff
@@ -235,7 +232,9 @@ protected function renameAccount() {
 			if ( $affectedRows ) {
 				$dbw->commit();
 				$this->addLog( "Changed user {$this->mOldUsername} to {$this->mNewUsername} in {$wgExternalSharedDB}" );
-				wfProfileOut( __METHOD__ );
+
+				User::clearUserCache( $this->mUserId );
```

Comment:
This is the real bug of this PR; there is code a little later on in execution that calls ```User::invalidateCache()``` on a user object that has been constructed with the new username.  Every single function in the ```User``` class seems to call ```User::load```, including this one, which will try to load the rest of the user data.  In this case it makes a direct DB query to get the user ID from the (new) user name.  Since we keep the user ID the same for the user, this new name now gets this old ID (where as the old name is set to a new row in the user table with a new ID).  As load continues, it tries to get the user data from memcached, by ID.  If its a hit, memcached then returns OLD DATA INCLUDING THE OLD USERNAME LIKE A COMPLETE DOPE.

Now the old username is mixing about with the new user name and eventually we get a ```duplicate key``` error when saving settings to the DB later on, which inexplicably saves the user object again, because its trying to write a row with the old user name and the old user ID, when there already is a row with that old username (the new record created at the beginning of the process).